### PR TITLE
Project: To-Do List 

### PR DESCRIPTION
**Pull request by tanzila-abedin**
A simple todo app to display the following : 

1. view all projects
2.  view all todos in each project (with title, description, priority, and due date)
3. Generate  a single todo to display its details
4. delete a todo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "todo-list",
+  "version": "1.0.0",
+  "description": "Js Project",
+  "private": true,
+  "scripts": {
```
Comment (rindrajosia) [inline package.json:6]: - [x] [REQUIRED] **Development** command should be added inside these `scripts` as it is part of **Necessary run common commands** 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))

File: package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack --progress --mode=production",
+    "watch": "webpack --progress --watch",
+    "server": "webpack-dev-server --open"
```
Comment (rindrajosia) [inline package.json]: - [x] [REQUIRED] I got an error when I use this command. It will be nice if you fix it:

File: src/modules/show.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    projectArray.forEach((element) => {
+      const { title } = element;
+      Show.newProjectTemplate(title);
+    });
+  }
+
+  static defaultProject() {
+    const project = new Project('Default');
+    Project.addProject(project);
+    // Show.newProjectTemplate("Default");
```
Comment (rindrajosia) [inline src/modules/show.js]: - [x] [REQUIRED]  Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [x] Files not related to the project should not be committed and added to .gitignore (e.g. .vscode, [.gitbub/workflows](https://github.com/tanzila-abedin/Todo-app/tree/feature/.gitbub/workflows)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] User should be able to edit the details of a to-do item.
- [x] Todos that I created disappears when I click other projects:


- [x] Do not use **alert()**. It’s ugly, impossible to style, stop code execution and display differently on different browsers. Use custom modal instead. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/alert_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/javascript-module-v-10/166782/path/step/97634928/))
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## To highlit :smile: 

- User can edit tasks

 ## Required Changes ♻️

- [ ] When I create a new project, tasks that I created previously in other projects disappear (LocalStorage and Interface):



_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] It will be nice if you reset the form once the task is created

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by tanzila-abedin** [comment]
Hi @rindrajosia thank you for the review. I have been trying to reproduce the error, but everything seems to be fine on my end. I am being able to create a project and change between projects and the tasks of each project would still appear when clicked on a specific project.

**Review by bellom — APPROVED** [review]
## Hi @tanzila-abedin  :clap: 

**Great work making all the necessary corrections from the reviews.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best**

**Pull request merged**